### PR TITLE
📱 Two-factor login views 

### PR DESCRIPTION
**Pull request by rowanseymour**
We still need to improve the views on the org home page for managing backup tokens, but since this is all beta gated, no harm in adding these views now.

Captura de Pantalla 2021-02-10 a la(s) 12 29 57

Captura de Pantalla 2021-02-11 a la(s) 11 59 30

Captura de Pantalla 2021-02-11 a la(s) 11 59 39

### CODE REVIEW

File: temba/orgs/views.py [hunk 1]
```diff
@@ -344,6 +349,95 @@ class Meta:
...
+            redirect_url = self.get_redirect_url()
+            if redirect_url:
+                verify_url += f"?{self.redirect_field_name}={urlquote(redirect_url)}"
+
+            return HttpResponseRedirect(verify_url)
+
+        return super().form_valid(form)
+
+
+class BaseTwoFactorView(AuthLoginView):
```
Comment (rowanseymour) [inline temba/orgs/views.py:372]: extending the `auth` package's login view gives us all their logic around success URLs

**Review by norkans7 — APPROVED** [review]

File: templates/orgs/login/login.haml [hunk 2]
```diff
@@ -1,5 +1,4 @@
--extends "smartmin/users/login_base.haml"
-
+-extends "orgs/login/login_base.haml"
```
Comment (rowanseymour) [inline templates/orgs/login/login.haml:1]: now that we're overriding the smartmin login view.. might as well have all our login templates in the same place

File: temba/orgs/views.py [hunk 3]
```diff
@@ -344,6 +349,102 @@ class Meta:
...
+            self.request.session[TWO_FACTOR_USER_SESSION_KEY] = str(user.id)
+
+            verify_url = reverse("users.two_factor_verify")
+            redirect_url = self.get_redirect_url()
+            if redirect_url:
+                verify_url += f"?{self.redirect_field_name}={urlquote(redirect_url)}"
+
+            return HttpResponseRedirect(verify_url)
+
+        return super().form_valid(form)
```
Comment (rowanseymour) [inline temba/orgs/views.py:371]: this is where the user would become authenticated, but if they have two factor enabled, that won't happen just yet

**Comment by nicpottier** [comment]
Glancing at https://github.com/Bouke/django-two-factor-auth for inspiration  I see a few things we don't do.. (why did we build this from scratch again?)

1. we never expire a login that is part way through
2. I don't see us having a limit to your number of valid second factor

Otherwise this looks good to me, but ya, I'll be honest I have some reservations about us missing some weird attach vector here..

**Review by nicpottier — APPROVED** [review]

**Comment by rowanseymour** [comment]
Attempt limit on background tokens seems.. important. Will look at adding that.

**Pull request merged**